client/src/components/Search/Search.js

```diff
@@ -1,68 +1,99 @@
-import React, { useState } from "react";
+import React, { useLayoutEffect, useState, useRef } from "react";
 import { Controller, useForm } from "react-hook-form";
-import "react-datepicker/dist/react-datepicker.css";
 import Calendar from "./Calendar";
+import Guests from "./Guests";
+import AdultsChildren from "./AdultsChildren";
+import { useHistory } from "react-router-dom";
+import { formatISO } from "date-fns";
+import addDays from "date-fns/addDays";
 
-const Search = () => {
-  const { handleSubmit, errors, control } = useForm();
-  const onSubmit = (data) => console.log(data);
+const Search = ({ initialState }) => {
+  const { handleSubmit, control } = useForm({
+    defaultValues: {
+      adults: initialState?.adults || 0,
+      children: initialState?.children || 0,
+    },
+  });
+  let history = useHistory();
```

Comment:
I would use `const` everywhere until you really have to change value of a variable